### ISSUE TITLE
Task 5:

### ISSUE THREAD

**Issue by mjenrungrot** [mjenrungrot/vpn#5]
- [x] Authenticate VPN Server
- - [x] Get a public-key certificate from a CA  334e84b04758f909be6ac7d5fb2980c7a8518cbe
- - [x] Add a code for making SSL connection to the server  d59df4ad73b9c9029510133dbae17b10d27304eb
- [x] Authenticate VPN Client
- - [x] Get a public-key certificate from a CA 334e84b04758f909be6ac7d5fb2980c7a8518cbe
- - [x] Add a code for sending certificate to the server d59df4ad73b9c9029510133dbae17b10d27304eb
- - [x] Add a code for verifying clients' certificate on the server d59df4ad73b9c9029510133dbae17b10d27304eb
- [x] Key Exchange 6e09d6805a80345ef16ea4ae907c99d7d371fa6c
- - [x] Use TCP for control channel
- - [x] Use UDP for sending VPN's data from TUN interface
- - [x] Send the session key from client to server
- - [x] Send the initial vector from client to server
- [x] Dynamic Reconfiguration
- - [x] Use fork() to control communications between TCP and UDP and signal interrrupt.
- - [x] Change the session key on the client end, and inform the server to make the similar change.
- - [x] Change the IV on the client end, and inform the server to make the similar change.
- - [x] Break the current VPN tunnel. The server needs to be informed, so it can release the corresponding
resources.

**Comment by mjenrungrot** [comment]
@famendola1 For the Task 5.3 Key Exchange, it mentions something about using TCP connection for the control channel between the client and the server. But our VPN uses UDP connection, so I'm confused if we have to make two connections, one for TCP and another one for UDP.

Note that I have added the code for authentication using certificates on both sides already. 

Edit1: Is it possible to use the UDP connection we created for the control channel? I looked online for a little bit, and it seems possible to use UDP connections for SSL stuff.
